File: Day-02/python/RainbowDashLabs/day_2.py
values = [(e.split(" ")[0], int(e.split(" ")[1])) for e in open("puzzle.txt")]

depth = 0
horizontal = 0

for k, v in values:
    if k == "forward":
        horizontal += v
    else:
        depth += v if k == "down" else -v

print(f"Part one {depth * horizontal}")


depth = 0
horizontal = 0
aim = 0

for k, v in values:
    if k == "forward":
        horizontal += v
        depth += aim * v
    else:
        aim += v if k == "down" else -v

print(f"Part two {depth * horizontal}")

File: Day-02/java/RainbowDashLabs/Day2.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.stream.Collectors;

public class Day2 {
    public static void main(String[] args) throws IOException {
        var values = Files.readAllLines(Path.of("puzzle.txt")).stream()
                .map(e -> new Entry(e.split(" ")[0], Integer.parseInt(e.split(" ")[1])))
                .collect(Collectors.toList());

        var depth = 0;
        var horizontal = 0;

        for (var entry : values) {
            if ("forward".equals(entry.direction)) {
                horizontal += entry.amount;
            } else {
                depth += "down".equals(entry.direction) ? entry.amount : -entry.amount;
            }
        }

        System.out.println("Part one is " + (depth * horizontal));


        depth = 0;
        horizontal = 0;
        var aim = 0;

        for (var entry : values) {
            if ("forward".equals(entry.direction)) {
                horizontal += entry.amount;
                depth += aim  * entry.amount;
            } else {
                aim += "down".equals(entry.direction) ? entry.amount : -entry.amount;
            }
        }

        System.out.println("Part two is " + (depth * horizontal));
    }

    private record Entry(String direction, int amount) {
    }
}
